### PR TITLE
Presentation style + refactoring of AT/ATS in PCA + Block ADFS

### PR DESCRIPTION
**Pull request by unpluggedk**
refactor AT and ATS, and add presentationStyle to the interactive token parameters.

Also, block ADFS authority from being created until support is fully tested.

### CODE REVIEW

File: MSAL/src/MSALInteractiveTokenParameters+Internal.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//
+//------------------------------------------------------------------------------
+
+#import "MSALInteractiveTokenParameters.h"
+#import "MSALTelemetryApiId.h"
+
+@interface MSALInteractiveTokenParameters ()
+
+- (id)internalParentController;
+- (NSInteger)internalPresentationStyle;
```
Comment (antrix1989) [inline MSAL/src/MSALInteractiveTokenParameters+Internal.h]: Seems we don't need these 2 internal methods. We can use directly `.parentViewController` and `. presentationStyle`

File: MSAL/src/public/MSALInteractiveTokenParameters.h [hunk 2]
```diff
@@ -70,6 +70,18 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (instancetype)initWithScopes:(NSArray<NSString *> *)scopes NS_DESIGNATED_INITIALIZER;
 
+#if TARGET_OS_IPHONE
+/*!
+ Modal presentation type for displaying authentication web content.
```
Comment (antrix1989) [inline MSAL/src/public/MSALInteractiveTokenParameters.h]: Should this be above `presentationStyle`?

File: MSAL/src/public/MSALInteractiveTokenParameters.h [hunk 3]
```diff
@@ -70,6 +70,18 @@ NS_ASSUME_NONNULL_BEGIN
...
+ */
+@property (nullable, weak, nonatomic) UIViewController *parentViewController;
+@property (nonatomic) UIModalPresentationStyle presentationStyle;
+#endif
+
+/*!
+ A specific prompt type for the interactive authentication flow.
```
Comment (antrix1989) [inline MSAL/src/public/MSALInteractiveTokenParameters.h]: Should this documentation be above `webviewType`?

File: MSAL/src/public/MSALInteractiveTokenParameters.h (hunk 2 above)
Comment (unpluggedk) [reply MSAL/src/public/MSALInteractiveTokenParameters.h]: Sure

File: MSAL/src/MSALInteractiveTokenParameters+Internal.h (hunk 1 above)
Comment (unpluggedk) [reply MSAL/src/MSALInteractiveTokenParameters+Internal.h]: let's do that.

Comment (unpluggedk) [reply MSAL/src/MSALInteractiveTokenParameters+Internal.h]: I did it by removing one of the abstract layers for the call internally.

File: MSAL/src/MSALPublicClientApplication.m [hunk 4]
```diff
@@ -930,98 +770,196 @@ - (void)acquireTokenSilentForScopes:(NSArray<NSString *> *)scopes
...
+    msidParams.accountIdentifier = parameters.account.lookupAccountIdentifier;
+    msidParams.validateAuthority = shouldValidate;
+    msidParams.extendedLifetimeEnabled = self.internalConfig.extendedLifetimeEnabled;
+    msidParams.clientCapabilities = self.internalConfig.clientApplicationCapabilities;
+    msidParams.extraURLQueryParameters = self.internalConfig.extraQueryParameters.extraURLQueryParameters;
+    msidParams.extraTokenRequestParameters = self.internalConfig.extraQueryParameters.extraTokenURLParameters;
+    msidParams.tokenExpirationBuffer = self.internalConfig.tokenExpirationBuffer;
+    msidParams.claimsRequest = parameters.claimsRequest.msidClaimsRequest;
+    
+    if (msidParams.validateAuthority
```
Comment (oldalton) [inline MSAL/src/MSALPublicClientApplication.m]: isn't this one present above already?

File: MSAL/src/instance/MSALADFSAuthority.m [hunk 5]
```diff
@@ -33,9 +33,18 @@
...
 
+#define ADFS_NOT_YET_SUPPORTED
+
 - (instancetype)initWithURL:(NSURL *)url
                       error:(NSError **)error
 {
+#ifdef ADFS_NOT_YET_SUPPORTED
+    if (error)
+    {
+        *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorUnsupportedFunctionality, @"AD FS authority is not supported yet in MSAL", nil, nil, nil, nil, nil);
```
Comment (oldalton) [inline MSAL/src/instance/MSALADFSAuthority.m]: should we return MSAL error here instead?

File: MSAL/src/public/MSALInteractiveTokenParameters.h [hunk 6]
```diff
@@ -70,8 +70,25 @@ NS_ASSUME_NONNULL_BEGIN
...
+#if TARGET_OS_IPHONE
+/*!
+ Modal presentation style for displaying authentication web content.
+ */
+@property (nullable, weak, nonatomic) UIViewController *parentViewController;
+@property (nonatomic) UIModalPresentationStyle presentationStyle;
+#endif
+
+/*!
+ A specific prompt type for the interactive authentication flow.
```
Comment (oldalton) [inline MSAL/src/public/MSALInteractiveTokenParameters.h]: nit: let's update documentation from prompt type to WebView type :)

**Pull request merged**